File: backend/main_api.py
'''
Lambda function that powers the main api
'''
import json, pymysql
import pymysql.cursors

def lambda_handler(event, context):
    connection = pymysql.connect(host='', user='', password='', database='',    cursorclass=pymysql.cursors.DictCursor)
    print(connection)
    
    with connection:
        with connection.cursor() as cursor:
            selectBasementDataSQL = "select temp,humd,UNIX_TIMESTAMP(date) * 1000 as 'unixTime' ,convert_tz(date,'+00:00','-05:00') as 'normalTime' from tempdata2 order by id desc limit 1"
            cursor.execute(selectBasementDataSQL)
            basementData = cursor.fetchall()

            selectBedroomDataSQL = "select temp,humd,UNIX_TIMESTAMP(date) * 1000 as 'unixTime' ,convert_tz(date,'+00:00','-05:00') as 'normalTime' from tempdata3 order by id desc limit 1"
            cursor.execute(selectBedroomDataSQL)
            bedroomData = cursor.fetchall()
            basementJsonData = basementData[0]['temp'],basementData[0]['humd'], basementData[0]['unixTime'], basementData[0]['normalTime']
            bedroomJsonData = bedroomData[0]['temp'],bedroomData[0]['humd'], bedroomData[0]['unixTime'], bedroomData[0]['normalTime']


            
    return {
    "basementData" : {
        "temp": basementData[0]['temp'],
         "humid": basementData[0]['humd'],
         "last_updated": basementData[0]['unixTime'], 
         "last_updated_normal;" :  json.dumps(basementData[0]['normalTime'], default=str)
    },
    "bedroomData": {
      "temp": bedroomData[0]['temp'],
         "humid": bedroomData[0]['humd'],
         "last_updated": bedroomData[0]['unixTime'],  
         "last_updated_normal;" :  json.dumps(bedroomData[0]['normalTime'], default=str)  
    }
    
    
    

    }

File: backend/main_api.go
package main

import (
	"log"
	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
	"database/sql"
	_ "github.com/go-sql-driver/mysql"
)

type SensorData struct{
	Temp string `json:"temp"`
	Humid string `json:"humid"`
	UnixTime string `json:"unixTime"`
	NormalTime string `json:"normalTime"`
}

var (
	temp string
	humid string
	unixTime string
	normalTime string
	firstRow *sql.Rows
	secondRow *sql.Rows
	sensorDataList []SensorData
)

func Handler(request events.APIGatewayProxyRequest) (map[string]SensorData, error) {
	sensorDataList = nil

	db, err := sql.Open("mysql",
	"admin:pass(localhost:3306)/sensor_data")
	if err != nil {
	   log.Fatal(err)
	}
	
	defer db.Close()
	log.Printf("Processing Lambda request %s\n", request.RequestContext.RequestID)

	firstRow, err := db.Query("select temp,humd,UNIX_TIMESTAMP(date) * 1000 as 'unixTime' ,convert_tz(date,'+00:00','-05:00') as 'normalTime' from tempdata2 order by id desc limit 1")
	for firstRow.Next() {
		err := firstRow.Scan(&temp, &humid, &unixTime, &normalTime)
		if err != nil {
			log.Fatal(err)
		}
		sensorDataList = append(sensorDataList, SensorData{Temp:temp,Humid:humid, UnixTime:unixTime, NormalTime:normalTime})
		log.Println(sensorDataList)

	}

	secondRow, err := db.Query("select temp,humd,UNIX_TIMESTAMP(date) * 1000 as 'unixTime' ,convert_tz(date,'+00:00','-05:00') as 'normalTime' from tempdata3 order by id desc limit 1")
	for secondRow.Next() {
		err := secondRow.Scan(&temp, &humid, &unixTime, &normalTime)
		if err != nil {
			log.Fatal(err)
		}
		sensorDataList = append(sensorDataList, SensorData{Temp:temp,Humid:humid, UnixTime:unixTime, NormalTime:normalTime})
	}

	return map[string]SensorData{"basementData": sensorDataList[0], "bedroomData": sensorDataList[1]}, nil

}

func main() {
	lambda.Start(Handler)
}
